src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -341,6 +346,15 @@ DistOpsValidityState(Node *node, const DistributeObjectOps *ops)
 		bool isPostprocess = false;
 		List *objectAddresses = ops->address(node, missingOk, isPostprocess);
 
+		if (HasDropCommandViolatesOwnership(node))
```

Comment:
Why does this require `ops->address`? Can't we move this as part of the above `else if` structure?